mux_test.go

```diff
@@ -0,0 +1,355 @@
+package e5e_test
```

Comment:
The tests are currently failing in CI due to a race condition. This needs to be fixed.